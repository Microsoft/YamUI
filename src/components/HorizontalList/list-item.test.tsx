/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ListItem, NestableBaseComponentProps } from '.';

describe('<ListItem />', () => {
  let component: ShallowWrapper<NestableBaseComponentProps, {}>;

  describe('without props', () => {
    beforeEach(() => {
      component = shallow(<ListItem />);
    });

    it('renders an li element', () => {
      expect(component.find('li').length).toBe(1);
    });

    it('renders its base className', () => {
      expect(component.hasClass('y-horizontalList--item')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<ListItem className="TEST_CLASSNAME" />);
    });

    it('includes that className', () => {
      expect(component.hasClass('TEST_CLASSNAME')).toBe(true);
    });

    it('still includes its base className', () => {
      expect(component.hasClass('y-horizontalList--item')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with child content', () => {
    beforeEach(() => {
      component = shallow(<ListItem>list item content</ListItem>);
    });

    it('renders the child content', () => {
      expect(component.text()).toBe('list item content');
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
