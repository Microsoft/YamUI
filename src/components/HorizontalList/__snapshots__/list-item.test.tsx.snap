// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> with additional className matches its snapshot 1`] = `
<li
  className="y-horizontalList--item TEST_CLASSNAME"
/>
`;

exports[`<ListItem /> with child content matches its snapshot 1`] = `
<li
  className="y-horizontalList--item"
>
  list item content
</li>
`;

exports[`<ListItem /> without props matches its snapshot 1`] = `
<li
  className="y-horizontalList--item"
/>
`;
